sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShimLoader.scala

```diff
@@ -414,4 +415,9 @@ object ShimLoader extends Logging {
   def loadGpuColumnVector(): Class[_] = {
     ShimReflectionUtils.loadClass("com.nvidia.spark.rapids.GpuColumnVector")
   }
+
+  def newAdaptiveSparkPlanHelperShim(): AdaptiveSparkPlanHelper =
+    ShimReflectionUtils.newInstanceOf[AdaptiveSparkPlanHelper](
+      "org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanHelper"
+    )
```

Comment:
since this class creates a compile-time reference to `AdaptiveSparkPlanHelper` via the import statement and the return type in the method signature I think it has the same effect as creating simply creating a val in ExecutionPlanCaptureCallback without messing with reflrection

```
private val sparkPlanHelper = new AdaptiveSparkPlanHelper()
```